### PR TITLE
Export layr iterrows APIs to cortex/coreapi

### PR DESCRIPTION
**Pull request by jnwatson**

### CODE REVIEW

File: synapse/cortex.py [hunk 1]
```diff
@@ -835,6 +835,94 @@ async def syncIndexEvents(self, matchdef, offsdict=None, wait=True):
         async for item in self.cell.syncIndexEvents(matchdef, offsdict=offsdict, wait=wait):
             yield item
 
+    async def iterFormRows(self, layriden, form, stortype=None, startvalu=None):
+        '''
+        Yields buid, valu tuples of nodes of a single form, optionally (re)starting at startvalue
+
+        Args:
+            layr (str):  Iden of the layer to retrieve the nodes
+            prop (str):  A universal property name.
```
Comment (Cisphyx) [inline synapse/cortex.py:844]: wrong arg here

File: synapse/cortex.py [hunk 2]
```diff
@@ -835,6 +835,94 @@ async def syncIndexEvents(self, matchdef, offsdict=None, wait=True):
...
+        async for item in self.cell.iterFormRows(layriden, form, stortype=stortype, startvalu=startvalu):
+            yield item
+
+    async def iterPropRows(self, layriden, form, prop, stortype=None, startvalu=None):
+        '''
+        Yields buid, valu tuples of nodes with a particular secondary property, optionally (re)starting at startvalue
+
+        Args:
+            layriden (str):  Iden of the layer to retrieve the nodes
+            prop (str):  A universal property name.
```
Comment (Cisphyx) [inline synapse/cortex.py:860]: args need updating here too

File: synapse/cortex.py (hunk 1 above)
Comment (Cisphyx) [reply synapse/cortex.py:844]: also missing `stortype` in the arg list

File: synapse/cortex.py [hunk 3]
```diff
@@ -835,6 +835,94 @@ async def syncIndexEvents(self, matchdef, offsdict=None, wait=True):
...
+            yield item
+
+    async def iterUnivRows(self, layriden, prop, stortype=None, startvalu=None):
+        '''
+        Yields buid, valu tuples of nodes with a particular universal property, optionally (re)starting at startvalue
+
+        Args:
+            layriden (str):  Iden of the layer to retrieve the nodes
+            prop (str):  A universal property name.
+            startvalu (Any):  The value to start at.  May only be not None if stortype is not None.
```
Comment (Cisphyx) [inline synapse/cortex.py:877]: missing stortype in the arg list

File: synapse/cortex.py [hunk 4]
```diff
@@ -835,6 +835,94 @@ async def syncIndexEvents(self, matchdef, offsdict=None, wait=True):
...
+    async def iterTagPropRows(self, layriden, tag, prop, form=None, stortype=None, startvalu=None):
+        '''
+        Yields (buid, valu) that match a tag:prop, optionally (re)starting at startvalu.
+
+        Args:
+            layriden (str):  Iden of the layer to retrieve the nodes
+            tag (str):  tag name
+            prop (str):  prop name
+            form (Optional[str]):  optional form name
+            stortype (Optional[int]): a STOR_TYPE_* integer representing the type of form:prop
```
Comment (Cisphyx) [inline synapse/cortex.py:916]: missing startvalu in the arg list

File: synapse/cortex.py [hunk 5]
```diff
@@ -4403,6 +4491,108 @@ async def _disableMigrationMode(self):
         if self.conf.get('trigger:enable'):
             self.trigson = True
 
+    async def iterFormRows(self, layriden, form, stortype=None, startvalu=None):
+        '''
+        Yields buid, valu tuples of nodes of a single form, optionally (re)starting at startvalu.
+
+        Args:
+            layr (str):  Iden of the layer to retrieve the nodes
+            prop (str):  A universal property name.
```
Comment (Cisphyx) [inline synapse/cortex.py:4500]: wrong arg, missing stortype

File: synapse/cortex.py [hunk 6]
```diff
@@ -4403,6 +4491,108 @@ async def _disableMigrationMode(self):
...
+        async for item in layr.iterFormRows(form, stortype=stortype, startvalu=startvalu):
+            yield item
+
+    async def iterPropRows(self, layriden, form, prop, stortype=None, startvalu=None):
+        '''
+        Yields buid, valu tuples of nodes with a particular secondary property, optionally (re)starting at startvalu.
+
+        Args:
+            layriden (str):  Iden of the layer to retrieve the nodes
+            prop (str):  A universal property name.
```
Comment (Cisphyx) [inline synapse/cortex.py:4519]: needs form and stortype args added

File: synapse/cortex.py [hunk 7]
```diff
@@ -4403,6 +4491,108 @@ async def _disableMigrationMode(self):
...
+            yield item
+
+    async def iterUnivRows(self, layriden, prop, stortype=None, startvalu=None):
+        '''
+        Yields buid, valu tuples of nodes with a particular universal property, optionally (re)starting at startvalu.
+
+        Args:
+            layriden (str):  Iden of the layer to retrieve the nodes
+            prop (str):  A universal property name.
+            startvalu (Any):  The value to start at.  May only be not None if stortype is not None.
```
Comment (Cisphyx) [inline synapse/cortex.py:4539]: needs stortype arg

File: synapse/cortex.py [hunk 8]
```diff
@@ -4403,6 +4491,108 @@ async def _disableMigrationMode(self):
...
+    async def iterTagPropRows(self, layriden, tag, prop, form=None, stortype=None, startvalu=None):
+        '''
+        Yields (buid, valu) that match a tag:prop, optionally (re)starting at startvalu.
+
+        Args:
+            layriden (str):  Iden of the layer to retrieve the nodes
+            tag (str):  tag name
+            prop (str):  prop name
+            form (Optional[str]):  optional form name
+            stortype (Optional[int]): a STOR_TYPE_* integer representing the type of form:prop
```
Comment (Cisphyx) [inline synapse/cortex.py:4584]: needs startvalu arg

File: synapse/lib/layer.py [hunk 9]
```diff
@@ -2224,6 +2224,16 @@ async def iterNodeEdgesN2(self, buid, verb=None):
             yield verb, s_common.ehex(n1buid)
 
     async def iterFormRows(self, form, stortype=None, startvalu=None):
+        '''
+        Yields buid, valu tuples of nodes of a single form, optionally (re)starting at startvalu.
+
+        Args:
+            prop (str):  A universal property name.
```
Comment (Cisphyx) [inline synapse/lib/layer.py:2231]: wrong arg

File: synapse/lib/layer.py [hunk 10]
```diff
@@ -2234,6 +2244,16 @@ async def iterFormRows(self, form, stortype=None, startvalu=None):
             yield item
 
     async def iterPropRows(self, form, prop, stortype=None, startvalu=None):
+        '''
+        Yields buid, valu tuples of nodes with a particular secondary property, optionally (re)starting at startvalu.
+
+        Args:
+            prop (str):  A universal property name.
```
Comment (Cisphyx) [inline synapse/lib/layer.py:2251]: wrong args

File: synapse/lib/layer.py [hunk 11]
```diff
@@ -2245,8 +2265,14 @@ async def iterPropRows(self, form, prop, stortype=None, startvalu=None):
 
     async def iterUnivRows(self, prop, stortype=None, startvalu=None):
         '''
+        Yields buid, valu tuples of nodes with a particular universal property, optionally (re)starting at startvalu.
+
         Args:
-            startvalu (Any): The value to start at.  May only be not None if stortype is not None.
+            prop (str):  A universal property name.
+            startvalu (Any):  The value to start at.  May only be not None if stortype is not None.
```
Comment (Cisphyx) [inline synapse/lib/layer.py:2272]: missing stortype arg

File: synapse/lib/layer.py [hunk 12]
```diff
@@ -2307,10 +2335,13 @@ async def iterTagRows(self, tag, form=None, starttupl=None):
...
         '''
-        Yields (buid, valu) that match a tag:prop
+        Yields (buid, valu) that match a tag:prop, optionally (re)starting at startvalu.
 
         Args:
-            form:  may be None
+            tag (str):  tag name
+            prop (str):  prop name
+            form (Optional[str]):  optional form name
+            stortype (Optional[int]): a STOR_TYPE_* integer representing the type of form:prop
```
Comment (Cisphyx) [inline synapse/lib/layer.py:2344]: missing startvalu arg

File: synapse/tests/test_cortex.py [hunk 13]
```diff
@@ -5177,3 +5178,83 @@ async def test_tag_model(self):
...
+            self.eq(ivals, tuple(sorted([row[1] for row in rows])))
+
+            # iterFormRows
+            await self.agenraises(s_exc.NoSuchLayer, prox.iterFormRows(badiden, 'inet:ipv4'))
+
+            rows = await alist(prox.iterFormRows(layriden, 'inet:ipv4'))
+            self.eq([(buid1, 1), (buid2, 2), (buid3, 3)], rows)
+
+            # iterTagRows
+            expect = sorted(
```
Comment (Cisphyx) [inline synapse/tests/test_cortex.py:5236]: seems to be missing the successful iterTagRows test that matches these values

Comment (jnwatson) [reply synapse/tests/test_cortex.py:5236]: Ok added a couple of tests that use it.

**Review by Cisphyx — APPROVED** [review]

**Pull request merged**